CMakeLists.txt

```diff
@@ -103,6 +103,14 @@ elseif(CMAKE_CXX_COMPILER_ID MATCHES Clang)
   add_compile_options(-fcolor-diagnostics)
 endif()
 
+# Allow selection of subcontroller for conditional build
+set(SUBCONTROLLER
+    "OpenCR"
+    CACHE STRING "Select what subcontroller to build for"
+)
+# Set the possible values of subcontroller for cmake-gui
+set_property(CACHE SUBCONTROLLER PROPERTY STRINGS "OpenCR" "cm740" "NUSense")
+
```

Comment:
I'd much rather change one cmake variable than go through a heap of roles and probably miss some if I'm doing it on the fly. As for keeping roles for both, they would be completely duplicated apart from the subcontroller and that just makes it less likely that they will all be kept up to date. I imagine this would be more likely to be used at robocup, if we need to go back to the CM740 for whatever reason, and that's already a high pressure environment where things can easily go wrong. I really don't see any downside of having a quick flag to change if we swap subcontrollers.

Reply:
I agree with @Bidski that this is not the right way to do this, it's gonna cause chaos.

I also do see your point that changing the subcontroller shouldn't result in duplicated roles all over the place.
My suggestion for this is take advantage of the fact that .role files are still cmake files and declare your hardware io etc like this.

```cmake
nuclear_role(
    ...
    platform::${NUBOTS_PLATFORM)::HardwareIO
)
```

Then have the platform-specific modules for each in that directory and you can select which module you use by changing that variable (set it up earlier in the same way we set up the choices for the build mode)

Reply:
That's cleaner, I didn't think about the fact that roles are just cmake files.
Have updated it.